src/fp-paradigm/js/main.js

```diff
@@ -36,6 +41,57 @@ const render = (form) => {
 const compose = (...functions) => (value = null) =>
     functions.reduceRight((x, f) => f(x), value);
 
-const main = (inputs) => compose(render, renderForm, renderFormGroups)(inputs);
+const main = (inputs, submitHandler) => {
+    compose(render, renderForm, renderFormGroups)(inputs);
+
+    const form = getNodeBySelector('.form');
+    form.addEventListener('submit', submitHandler);
+};
+
+const getNodeBySelector = (selector) => document.querySelector(selector);
+const getValueFromInput = (selector) => document.querySelector(selector).value;
+const deleteTodo = (i) =>
+    dispatch({
+        type: 'DELETE_TODO',
+        id: i,
+    });
+const toggleTodo = (i) =>
+    dispatch({
+        type: 'TOGGLE_TODO',
+        id: i,
+    });
+
+const rerender = (todos) => {
+    const todosView = getNodeBySelector('.todos');
+    const taskInput = getNodeBySelector('.task-input');
+
+    taskInput.value = '';
+
+    const newTodosList = todos.reduce(
+        (nodes, todo, i) =>
+            (nodes += `
+            <div class="todo">
+                <div class="todo__id">${todo.id}</div>
+                <div class="todo__task">${todo.task}</div>
+                <div class="todo__completed">${todo.completed}</div>
+                <button onclick="deleteTodo(${i})")">Delete</button>
+                <button onclick="toggleTodo(${i})")">Completed</button>
+            </div>
+        `),
+        '',
+    );
+
+    todosView.innerHTML = newTodosList;
+};
+
+const submitHandler = (e) => {
+    e.preventDefault();
+    const task = getValueFromInput('.task-input');
+
+    dispatch({
+        type: 'ADD_TODO',
+        task,
+    });
+};
 
-console.log(main(state.formGroups.inputs));
+main(state.formGroups.inputs, submitHandler);
```

Comment:
In the future, InshaAllah, i want to covert all of them to typescript and bundle them with help of webpack. And we could implement package `.env` which stores node environments, to which we can have in our codes. Also there should be covered some other parts, but after finishing this project, we have to move to other topics. InshaAllah.